### PR TITLE
 many-to-many relations: Mention that fields and references are forbidden in `@relation` for implicit m-n relations (+ improve related documentation)

### LINKED ISSUE #3214: Remove `references` syntax from implicit many-to-many relations
Currently Prisma allows the following syntax for implicit many-to-many relations, with `id` as the only valid value for `references` :

```
model First {
    id Int @id
    seconds Second[] @relation("my-relation", references: [id])
}

model Second {
    id String @id
    firsts First[] @relation("my-relation", references: [id])
}
```

This use of `references` is no longer going to be allowed, to make it clearer what can be changed – see [this PSL proposal](https://www.notion.so/prismaio/Proposal-Forbid-the-references-argument-in-relation-attributes-in-implicit-many-to-many-relations-5851708905324e6c8fb8b581e359717b). 

These three alternatives will still be valid:

```
model First {
    id Int @id
    seconds Second[]
}

model Second {
    id String @id
    firsts First[]
}
```

```
model First {
    id Int @id
    seconds Second[] @relation
}

model Second {
    id String @id
    firsts First[] @relation
}
```

```
model First {
    id Int @id
    seconds Second[] @relation("my-relation")
}

model Second {
    id String @id
    firsts First[] @relation("my-relation")
}
```

It looks like the current docs on [how to use implicit many-to-many relations](https://prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#rules-for-defining-an-implicit-m-n-relation) don't use the deprecated syntax anyway. So only need to check docs for any examples that use it and fix them.

### PR DESCRIPTION
**Pull request by janpio**
see comments

clsoes https://github.com/prisma/docs/issues/3214

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 1]
```diff
@@ -282,6 +282,7 @@ Implicit m-n relations:
 
 - Use a specific [convention for relation tables](#conventions-for-relation-tables-in-implicit-m-n-relations)
 - Do **not** require the `@relation` attribute unless you need to [disambiguate relations](/concepts/components/prisma-schema/relations#disambiguating-relations) (this is not specific to implicit n-m relations).
+- If they use an `@relation` attribute for disambiguation, you can still not use the `fields` and `references` arguments as these are implicit and fixed for implicit m-n relations.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Mention that `references` (and `fields`) are not allowed when you use `@relation` on implicit m:n

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 2]
```diff
@@ -293,17 +294,23 @@ Implicit m-n relations:
 
   </Admonition>
 
-#### Conventions for relation tables in implicit m-n-relations
+#### Conventions for relation tables in implicit m-n relations
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:297]: Unrelated headline fix

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 3]
```diff
@@ -293,17 +294,23 @@ Implicit m-n relations:
...
-- **Valid**: `_CategoryToPost`, `_MyRelation`
-- **Invalid**: `CategoryToPost`, `MyRelation`
+- It must start with an underscore `_`
+- Then follows the alphabetically lower table name (in this case `Category`)
+- After which `To` connects to the rest of the name
+- Which is the alphavetically higher table name  (in this case `Post`)
+
+In the example the correct table name is `_CategoryToPost`.
+
+(When creating an implicit m-n relation via Prisma itself, you can also [configure the name of the relation table to be different](#configuring-the-name-of-the-relation-table-in-implicit-many-to-many-relations) to be for example `_MyRelationTable`)
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: I think the description here was actually misleading until now. The intro for the Conventions section makes this about Introspection - and there you start with _no_ schema file and then is can only pick up table names with _exactly_ this name.

The description of the rules is probably terrible, so that should be rewritten in better English @keerlu.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 1 above)
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: The added line might actually be to loose still, and could be replaced with something like this depending on the result of https://prisma-company.slack.com/archives/C01TM7UM83Y/p1655918528824109

```suggestion
- Do **not** require the `@relation` attribute unless you need to [disambiguate relations](/concepts/components/prisma-schema/relations#disambiguating-relations) (this is not specific to implicit n-m relations). You can only use a nameless argument or the `name` argument in that case: `@relation("MyRelationTable")` or `@relation(name: "MyRelationTable")`. Other arguments like `fields`, `references`, `onDelete` or `onUpdate` are not possible as their values are implicit and fixed for implicit m-n relations. 
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx [hunk 4]
```diff
@@ -283,8 +283,8 @@ A many-to-many self-relation looks as follows:
 model User {
   id         Int     @id @default(autoincrement())
   name       String?
-  followedBy User[]  @relation("UserFollows", references: [id])
-  following  User[]  @relation("UserFollows", references: [id])
+  followedBy User[]  @relation("UserFollows")
+  following  User[]  @relation("UserFollows")
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx:287]: I think this was the only _actual_ broken implicit m:n example in the whole docs that was using `references` 😆

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx [hunk 5]
```diff
@@ -310,6 +310,7 @@ This relation expresses the following:
 - "a user can be followed by zero or more users"
 - "a user can follow zero or more users"
 
+TODO THis is only correct for relational databases, the MongoDB example is using lists and is Mongo's m:n - not really "implicit" (There is already a mention of this further down where it shows the SQL/data structure)
 Note that this many-to-many-relation is [implicit](many-to-many-relations#implicit-many-to-many-relations). This means Prisma maintains a [relation table](/concepts/components/prisma-schema/relations/many-to-many-relations#relation-tables) for it in the underlying database.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx]: Note for @keerlu

File: content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx [hunk 6]
```diff
@@ -228,6 +228,7 @@ This however tells Prisma to expect **two** separate one-to-many relationships.
 
 The following example is the correct way to define an implicit many-to-many relationship.
 
+TODO Does the below work? Or do we need to name both sides of the relation?
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx]: Question for @prisma/schema-team

File: content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx [hunk 7]
```diff
@@ -245,7 +246,7 @@ model Category {
 ```
 
 The `@relation` annotation can also be used to name the underlying relation table created on a implicit many-to-many relationship.
-
+TODO This deserves a link as we document that on the implicit m:n section of the m:n page.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx]: TODO for @keerlu

File: content/200-concepts/100-components/01-prisma-schema/06-relations/index.mdx [hunk 8]
```diff
@@ -281,12 +281,11 @@ model Category {
 </TabbedContent>
 
 > **Note**: This schema is the same as the [example data model](/concepts/components/prisma-schema/data-model) but has all [scalar fields](/concepts/components/prisma-schema/data-model#scalar-fields) removed (except for the required [relation scalars](/concepts/components/prisma-schema/relations#annotated-relation-fields-and-relation-scalar-fields)) so you can focus on the [relation fields](#relation-fields).
-> **Note**: Implicit many-to-many relations do **not** require the `@relation` attribute unless you need to [disambiguate relations](#disambiguating-relations) (this is not specific to implicit n-m relations).
+> **Note**: Implicit many-to-many relations do **not** require the `@relation` attribute unless you need to [disambiguate relations](#disambiguating-relations).
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/index.mdx:284]: The note is specific, so I think the note in parens behind it was weird/wrong.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 1 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: I've attempted to reword this and also include the information in the comment below, probably worth another check.

Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: See comment above

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 3 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Reworded this, again worth a check to see that it's still correct

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx (hunk 4 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx:287]: Ah so there was one! Didn't notice any on my very quick check before!

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 9]
```diff
@@ -281,8 +281,8 @@ const getPostsAndCategories = await prisma.post.findMany({
 Implicit m-n relations:
 
 - Use a specific [convention for relation tables](#conventions-for-relation-tables-in-implicit-m-n-relations)
-- Do **not** require the `@relation` attribute unless you need to [disambiguate relations](/concepts/components/prisma-schema/relations#disambiguating-relations) (this is not specific to implicit n-m relations).
-- If they use an `@relation` attribute for disambiguation, you can still not use the `fields` and `references` arguments as these are implicit and fixed for implicit m-n relations. 
+- Do **not** require the `@relation` attribute unless you need to [disambiguate relations](/concepts/components/prisma-schema/relations#disambiguating-relations) with a name, e.g. `@relation("MyRelationTable")` or `@relation(name: "MyRelationTable")`.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: thsi is not related to a table fundamentally, just defines the table name underneath as a side effect:
```suggestion
- Do **not** require the `@relation` attribute unless you need to [disambiguate relations](/concepts/components/prisma-schema/relations#disambiguating-relations) with a name, e.g. `@relation("MyRelation")` or `@relation(name: "MyRelation")`.
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 10]
```diff
@@ -281,8 +281,8 @@ const getPostsAndCategories = await prisma.post.findMany({
...
+  - If you do use the `@relation` attribute for disambiguation, you cannot use the `references`, `fields`, `onUpdate` or `onDelete` arguments. This is because these take a fixed value for implicit m-n relations and cannot be changed.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: You can never use the other arguments, not only if you use `@relation` for disambiguation.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx (hunk 4 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx:287]: It was a weird one with the self relation.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 1 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Left comments on the new version. Much better already, but not there yet.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 11]
```diff
@@ -299,18 +299,18 @@ Implicit m-n relations:
...
-- Which is the alphavetically higher table name (in this case `Post`)
+- Then the name of the first table in alphabetical order (in this case `Category`)
+- Then `To`
+- Then the name of the second table in alphabetical order (in this case `Post`)
 
-In the example the correct table name is `_CategoryToPost`.
+In the example, the correct table name is `_CategoryToPost`.
 
-(When creating an implicit m-n relation via Prisma itself, you can also [configure the name of the relation table to be different](#configuring-the-name-of-the-relation-table-in-implicit-many-to-many-relations) to be for example `_MyRelationTable`)
+When creating an implicit many-to-many relation yourself in the Prisma schema file, you can [configure the relation name](#configuring-the-name-of-the-relation-table-in-implicit-many-to-many-relations) to have a different name, for example `_MyRelationTable`.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: You configure the relation name, but that changes the name that the table gets when being migrated. So it is kind of indirect but with exactly that result.

(Do we also add the underscore to these custom implicit m:n table named @do4gr?)

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 3 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Yes much better. Left comment on detail on the new version.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx (hunk 5 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx]: Added that this note is for relational databases only

File: content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx (hunk 7 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx]: Done

File: content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx (hunk 6 above)
Comment (do4gr) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx]: Yeah, that won't work. There are multiple issues in that schema:
- there are duplicate relation fields on both models, (categories twice on Post and posts twice on Category)
- if you name a relation on one side, you also need to name it on the other one (you can have one unnamed relation fields pair though)

This would work: 
```
model Post {
   id         Int        @id @default(autoincrement())
  title      String
  categories Category[] @relation("Category")
  categories2 Category[]
}
model Category {
  id    Int    @id @default(autoincrement())
  name  String
  posts Post[] @relation("Category")
  posts2 Post[] 
}
```

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 11 above)
Comment (do4gr) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: You would give the name as `MyRelationTable` and we would then under the hood add the underscore to create a table named `_MyRelationTable`

File: content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx (hunk 6 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx]: It looks like the example has delete/add highlighting though? So there isn't a duplication of fields, it's removing one line and adding the other

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 11 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Thanks, clarified this.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx (hunk 6 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/500-troubleshooting-relations.mdx]: Removing TODO comment so I can take this PR out of draft. I'll keep this conversation unresolved for now though and update example if necessary.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 10 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: (Unresolving comment)

This is still unclear unfortunately and makes it sounds like this is a special case of when you disambiguate. But this is always the case.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx [hunk 12]
```diff
@@ -310,9 +310,7 @@ This relation expresses the following:
 - "a user can be followed by zero or more users"
 - "a user can follow zero or more users"
 
-Note that this many-to-many-relation is [implicit](many-to-many-relations#implicit-many-to-many-relations). This means Prisma maintains a [relation table](/concepts/components/prisma-schema/relations/many-to-many-relations#relation-tables) for it in the underlying database.
-
-If you need the relation to hold other fields, on relational databases you can create an [explicit](many-to-many-relations#explicit-many-to-many-relations) many-to-many self relation as well. The explicit version of the self relation shown previously is as follows:
+Note that for relational databases, this many-to-many-relation is [implicit](many-to-many-relations#implicit-many-to-many-relations). This means Prisma maintains a [relation table](/concepts/components/prisma-schema/relations/many-to-many-relations#relation-tables) for it in the underlying database. If you need the relation to hold other fields, you can create an [explicit](many-to-many-relations#explicit-many-to-many-relations) many-to-many self relation as well. The explicit version of the self relation shown previously is as follows:
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx]: Why remove the empty line here? Before it was clearer what is implicit (first paragraph) and explicit (second paragraph).

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 10 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: I've moved the bullet to top level and removed the 'for disambiguation' - hopefully clearer now

File: content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx (hunk 12 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/400-self-relations.mdx]: Good question. I've put it back
